helix-core/src/main/java/org/apache/helix/controller/common/CapacityNode.java

```diff
@@ -78,13 +95,13 @@ public String getId() {
    * @return The number of partitions currently assigned to this node
    */
   public int getCurrentlyAssigned() {
-    return _currentlyAssigned;
+    return _partitionMap.values().stream().mapToInt(Set::size).sum();
```

Comment:
Realtime compute could be slow... I know this could be better looking or clear logic but our target in computation performance should be high (ms level).

A pre computed value is better.